addons/mail/static/tests/owl/component/attachment_tests.js

```diff
@@ -201,51 +281,276 @@ QUnit.test('default: PDF', async function (assert) {
+            .length,
+        0,
+        "attachment should not have its filename shown");
+    assert.strictEqual(
+        attachment
+            .querySelectorAll(`
+                :scope
+                .o_Attachment_details
+                .o_Attachment_extension`)
+            .length,
+        1,
+        "attachment should have its extension shown");
+});
+
+QUnit.test('layout with card details and filename and extension', async function (assert) {
+    assert.expect(3);
+    await this.start();
+    const attachmentLocalId = this.store.dispatch('createAttachment', {
+        filename: "test.txt",
+        id: 750,
+        mimetype: 'text/plain',
+        name: "test.txt",
+    });
+    await this.createAttachment(attachmentLocalId, {
+        allowPreview: false,
+        detailsMode: 'card',
+        isDownloadable: false,
+        isEditable: false,
+        showExtensionInDetails: true,
+        showFilenameInDetails: true
+    });
     const attachment = document.querySelector('.o_Attachment');
-    assert.ok(
-        attachment.classList.contains('o-basic-layout'),
-        "attachment should use basic layout by default");
-    assert.ok(
-        attachment.classList.contains('o-viewable'),
-        "MP4 attachment should be viewable");
     assert.strictEqual(
-        attachment.title,
-        "test.mp4",
-        "attachment should have filename as title attribute");
+        attachment
+            .querySelectorAll(`
+                :scope
+                .o_Attachment_details`)
+            .length,
+        1,
+        "attachment should have a details part");
+    assert.strictEqual(
+        attachment
+            .querySelectorAll(`
+                :scope
+                .o_Attachment_details
+                .o_Attachment_filename`)
+            .length,
+        1,
+        "attachment should have its filename shown");
+    assert.strictEqual(
+        attachment
+            .querySelectorAll(`
+                :scope
+                .o_Attachment_details
+                .o_Attachment_extension`)
+            .length,
+        1,
+        "attachment should have its extension shown");
+});
+
+QUnit.test('simplest layout with hover details and filename and extension', async function (assert) {
+    assert.expect(8);
+    await this.start();
+    const attachmentLocalId = this.store.dispatch('createAttachment', {
+        filename: "test.txt",
+        id: 750,
+        mimetype: 'text/plain',
+        name: "test.txt",
+    });
+    await this.createAttachment(attachmentLocalId, {
+        allowPreview: false,
+        detailsMode: 'hover',
+        isDownloadable: true,
+        isEditable: true,
+        showExtensionInDetails: true,
+        showFilenameInDetails: true
+    });
+    const attachment = document.querySelector('.o_Attachment');
+    assert.strictEqual(
+        attachment
+            .querySelectorAll(`
+                :scope
+                > .o_Attachment_details`)
```

Comment:
Yeah but I need to check that I have this : 
```
.o_Attachment
    .o_Attachment_image
        .o_Attachment_details
```
and NOT this
```
.o_Attachment
    .o_Attachment_details
```

--> I'll test with the modifier I needed to add for your SCSS remarks to avoid the >